File: 2026/array/four_divisors_1390.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var sumFourDivisors = function (nums) {
    const solve = (num) => {
        let divisors = 0
        let sum = 0
        for (let div = 1; div * div <= num; div++) {
            if (num % div === 0) {
                let other = num / div
                if (div === other) {
                    divisors++
                    sum += div
                } else {
                    divisors += 2
                    sum += (div + other)
                }
            }
            if (divisors > 4) {
                return 0
            }
        }
        return divisors === 4 ? sum : 0
    }
    let result = 0
    for (const num of nums) {
        result += solve(num)
    }
    return result
};

File: 2026/array/four_divisors_1390.py
class Solution:
    def solve(self, num: int) -> int:
        divisors = 0
        total_sum = 0
        div = 1
        while div * div <= num:
            if num % div == 0:
                other = num // div
                if div == other:
                    divisors += 1
                    total_sum += div
                else:
                    divisors += 2
                    total_sum += (div + other)
            if divisors > 4:
                return 0
            div += 1
        return total_sum if divisors == 4 else 0

    def sumFourDivisors(self, nums: List[int]) -> int:
        result = 0
        for num in nums:
            result += self.solve(num)
        return result
